src/Surfnet/StepupMiddleware/ApiBundle/Authorization/Service/AuthorizationService.php

```diff
@@ -110,6 +121,51 @@ public function assertRegistrationOfSelfAssertedTokensIsAllowed(IdentityId $iden
         return $this->allow();
     }
 
+    /**
+     * Is an identity allowed to self vet using a self-asserted token?
+     *
+     * One is allowed to do so when:
+     *  - SAT is allowed for the institution of the identity
+     *  - All of the tokens of the identity are vetted using the SAT vetting type
+     */
+    public function assertSelfVetUsingSelfAssertedTokenIsAllowed(IdentityId $identityId): AuthorizationDecision
+    {
+        $identity = $this->identityService->find((string)$identityId);
+        if (!$identity) {
+            return $this->deny('Identity not found');
+        }
+
+        $institution = new Institution((string)$identity->institution);
+        $institutionConfiguration = $this->institutionConfigurationService
+            ->findInstitutionConfigurationOptionsFor($institution);
+        if (!$institutionConfiguration) {
+            return $this->deny('Institution configuration could not be found, unable to ascertain if self-asserted tokens feature is enabled');
+        }
+
+        if (!$institutionConfiguration->selfAssertedTokensOption->isEnabled()) {
+            return $this->deny(sprintf('Institution "%s", does not allow self-asserted tokens', (string) $identity->institution));
+        }
```

Comment:
Seems like a repetition of code (r171,r185). Perhaps use a separate method to fetch the institution configuration by the identity?